packages/url.lua

```diff
@@ -33,43 +87,90 @@ SILE.registerCommand("href", function (options, content)
       content)
   else
     options.src = content[1]
-    local breakpat = options.breakpat or "/"
-    content = inputfilter.transformContent(content, urlFilter, breakpat)
     SILE.call("pdf:link", { dest = options.src, external = true,
       borderwidth = options.borderwidth,
       borderstyle = options.borderstyle,
       bordercolor = options.bordercolor,
       borderoffset = options.borderoffset },
       function (_, _)
-        SILE.call("code", {}, content)
+        SILE.call("url", { language = options.language }, content)
       end)
   end
-end)
+end, "Inserts a PDF hyperlink.")
 
 SILE.registerCommand("url", function (options, content)
-  local breakpat = options.breakpat or "/"
-  local result = inputfilter.transformContent(content, urlFilter, breakpat)
-  SILE.call("code", {}, result)
-end)
+  SILE.settings.temporarily(function ()
+    local primaryPenalty = SILE.settings.get("url.linebreak.primaryPenalty")
+    local secondaryPenalty = SILE.settings.get("url.linebreak.secondaryPenalty")
+    local worsePenalty = primaryPenalty + secondaryPenalty
+
+    if options.language then
+      SILE.languageSupport.loadLanguage(options.language)
+      if options.language == "fr" then
```

Comment:
I understand French needs special handling here, but this doesn't feel like the right place to set this up. We're almost certain to run into other languages with similar needs, and having language specific decision trees in packages doesn't feel right.

I am okay with merging this as is to get the ball rolling, but I think we should consider refactoring this so that the language support modules have a generic flag of some kind for applying (or not applying) more invasive typographical details.

Right now we use `und` to disable hyphenation, but that also disables other language features. We should probably offer better control over this.

Reply:
I opened an issue #1242 for this topic.